src/Maestro/Microsoft.DotNet.Maestro.Tasks.Tests/GetManifestAsAssetTests.cs

```diff
@@ -0,0 +1,204 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using FluentAssertions;
+using Microsoft.DotNet.Maestro.Client.Models;
+using Microsoft.DotNet.Maestro.Tasks.Tests.Mocks;
+using NUnit.Framework;
+
+namespace Microsoft.DotNet.Maestro.Tasks.Tests
+{
+    [TestFixture]
+    public class GetManifestAsAssetTests
+    {
+        private PushMetadataToBuildAssetRegistry pushMetadata;
+        private const string locationString = "https://url.test/accountName/ProjectName/_apis/build/builds/buildNumberHere/artifacts";
+        private const string newManifestName = "NewManifest";
+
+        internal static readonly AssetData PackageAsset1 =
+            new AssetData(true)
+            {
+                Locations = ImmutableList.Create(
+                    new AssetLocationData(LocationType.Container)
+                    { Location = locationString }),
+                Name = "PackageAsset1Name",
+                Version = "1.2.3-beta.1235.6"
+            };
+
+        internal static readonly AssetData ExpectedPackageAsset1 =
+            new AssetData(true)
+            {
+                Locations = ImmutableList.Create(
+                    new AssetLocationData(LocationType.Container)
+                    { Location = locationString }),
+                Name = $"assets/manifests/thisIsARepo/1.2.3-beta.1235.6/{newManifestName}",
+                Version = "1.2.3-beta.1235.6"
+            };
+
+        [SetUp]
+        public void SetupGetManifestAsAssetTests()
+        {
+            pushMetadata = new PushMetadataToBuildAssetRegistry
+            {
+                getEnvProxy = new GetEnvMock()
```

Comment:
You could also use a Mock here, rather than implementing the class yourself.

Reply:
I don't have much experience with the different patterns for mocking. Is there an advantage to using that pattern when direct DI isn't an option? In general this class is going to return different values for different inputs, they just aren't needed for this set of tests.

Reply:
Mock is great for these. Check this out. You'd do the following: (doing this from memory so might have slight errors):

```
getEnvMock = new Mock<IGetEnvProxy>();
getEnvMock.Setup(s => s.GetEnv("MYENVVAR1")).Returns("ENVVAR1_VALUE"); // This says that "every time someone calls getEnvProxy.GetEnv with "MYENVVAR1", it should return "ENVVAR1_VALUE".
getEnvMock.Setup(s => s.GetEnv("MYENVVAR1")).Returns("ENVVAR2_VALUE"); // This says that "every time someone calls getEnvProxy.GetEnv with "MYENVVAR2", it should return "ENVVAR2_VALUE".
getEnvProxy = getEnvMock.Object; // This returns an IGetEnvProxy, the thing that Mock<IGetEnvProxy> is wrapping
```

Reply:
This is very slick, thanks for putting it together for me!